FastSimulation/Tracking/plugins/SimTrackIdProducer.h

```diff
@@ -26,13 +26,16 @@ class SimTrackIdProducer : public edm::stream::EDProducer <>
 
       virtual void produce(edm::Event& e, const edm::EventSetup& es) override;
 
-      std::vector<int> SimTrackIds;
 
 private:
 
       // consumes 
       edm::EDGetTokenT<reco::TrackCollection>  trackToken;
-
+      double maxChi2_;
+      std::vector< edm::EDGetTokenT<edm::ValueMap<int> > > overrideTrkQuals_;
+      bool filterTracks_ = false;
+      reco::TrackBase::TrackQuality trackQuality_;
+     
 };
 
 #endif
```

Comment:
I am not convinced that testing track quality criteria & chi2 should go into the simtrackid producer. This module should only provide a mask.